pilot/pkg/model/service.go

```diff
@@ -1287,6 +1287,20 @@ func (s *Service) GetAllAddressesForProxy(node *Proxy) []string {
 	return nil
 }
 
+// UnconditionallyGetAllAddressesForProxy returns a k8s service's extra addresses to the cluster where the node resides.
```

Comment:
This duplication kind of sucks, intent was to make it so ambient is always dualstack even without features.EnableDualStack. TBD